src/Messages/OWSMessageSender.m

```diff
@@ -281,9 +296,14 @@ - (void)tryWithRemainingRetries:(NSUInteger)remainingRetries
 {
     DDLogDebug(@"%@ remainingRetries: %lu", self.tag, (unsigned long)remainingRetries);
 
+    // Use this flag to ensure a given operation only succeeds or fails once.
+    __block BOOL onceFlag = NO;
     RetryableFailureHandler retryableFailureHandler = ^(NSError *_Nonnull error) {
         DDLogInfo(@"%@ Sending failed.", self.tag);
 
+        OWSAssert(!onceFlag);
```

Comment:
Is this redundant with the once-checking that happens in the handlers wrapped in init?